core/src/mast/serialization/mod.rs

```diff
@@ -232,35 +229,105 @@ impl Deserializable for MastForest {
             mast_forest
         };
 
+        let basic_block_decorators: Vec<(usize, DecoratorList)> = read_block_decorators(source)?;
+        for (node_id, decorator_list) in basic_block_decorators {
+            let node_id = MastNodeId::try_from((node_id, &mast_forest))?;
+            mast_forest.set_decorators(node_id, decorator_list);
```

Comment:
```suggestion
            match &mut mast_forest[node_id] {
                MastNode::Block(basic_block) => {
                  basic_block.init_decorators(decorator_list);
                },
                other => {
                    return Err(DeserializationError::InvalidValue(format!(
                        "expected mast node with id {node_id} to be a basic block, found {other:?}"
                    )))
                },
            }
```

I would prefer doing something like this rather than implement `set_decorators` on `MastForest` which really only supports basic blocks. If we have invalid serialized bytes and the `node_id` would point to a Join node for example, then `set_decorators` would silently ignore that this is an error. So we should explicitly handle it so deserialization fails if we encounter something like that.